### ISSUE TITLE
Visual Studio App Center Releases Failing

### ISSUE THREAD

**Issue by steve09014098** [badges/shields#9950]
### Are you experiencing an issue with...

shields.io

### 🐞 Description

Attempting to use the page [Visual Studio App Center Release](https://shields.io/badges/visual-studio-app-center-releases) to generate a version number for my application in App Center.

In [Reqbin](https://reqbin.com/) I can retrieve a valid JSON file using the URL ```https://api.appcenter.ms/v0.1/apps/#org/#app/releases/latest``` (note I took this URL from the source code in this repository)

I generated an organization level token and attempted to use my valid Owner, App and Token to generate the badge but I get the an error badge that says ```Release - Invalid Response Data```

I use exactly the same details on this page [App Center Size](https://shields.io/badges/visual-studio-app-center-size) and it returns the correct size of the application.

Therefore I am certain that the data is correct, the API call is correct, the token, app and owner are correct but I assume since this feature was added the JSON response has changed slightly causing this to fail.

I notice that the code seems to generate a badge from both the short and long versions, but my short version string is empty

The URL I use is: ```https://api.appcenter.ms/v0.1/apps/#org/#app/releases/latest```
In Reqbin I set a header of "x-api-token:#token"
Clicking Execute gives me the following:

```
{
    "app_name": "###",
    "app_display_name": "###,
    "app_os": "Windows",
    "app_icon_url": "###",
    "owner": {
        "name": "###",
        "display_name": "###"
    },
    "is_external_build": false,
    "origin": "appcenter",
    "id": 1,
    "version": "v1.0.13.0",
    "short_version": "",
    "size": 596325657,
    "min_os": null,
    "device_family": null,
    "bundle_identifier": "44b8699fd595efe748522870b71ddbec",
    "fingerprint": "5fd96e7ab8cc2bec40158d19f027c966",
    "uploaded_at": "2024-02-02T14:34:29.872Z",
    "download_url": "###",
    "install_url": "###",
    "mandatory_update": false,
    "enabled": true,
    "fileExtension": "zip",
    "is_latest": true,
    "release_notes": "###",
    "package_hashes": ["b60b0b0c21c64b6cc95903c9119a1707920ffbefecec2e44a95fcb4de056a746"],
    "build": {
        "commit_hash": "370ce5dfcb7ec889d88ff3171ef25d1804d414e8"
    },
    "destination_type": "group",
    "status": "available",
    "distribution_group_id": "00000000-0000-0000-0000-000000000000",
    "distribution_groups": [{
        "id": "00000000-0000-0000-0000-000000000000",
        "name": "Collaborators",
        "origin": "appcenter",
        "display_name": "Collaborators",
        "is_public": false
    }]
}
```

### 💡 Possible Solution

Maybe something to do with an empty short version field?

**Comment by chris48s** [comment]
>  get the an error badge that says `Release - Invalid Response Data`

Can you post the badge URL that is giving you this issue.
Cheers

**Comment by steve09014098** [comment]
Hi
Im not sure I want to provide the actual URL since it has proprietary information including a Personal Access Token for my App Center account.
However the URL is:

URL: ```https://img.shields.io/visual-studio-app-center/releases/version/MyOwner/MyApplication/MyToken```

Markdown: ```![Visual Studio App Center Releases](https://img.shields.io/visual-studio-app-center/releases/version/MyOwner/MyApplication/MyToken)```
rst: .. i```mage:: https://img.shields.io/visual-studio-app-center/releases/version/MyOwner/MyApplication/MyToken
   :alt: Visual Studio App Center Releases```

Ascii Doc: ```image:https://img.shields.io/visual-studio-app-center/releases/version/MyOwner/MyApplication/MyToken[Visual Studio App Center Releases]```

HTML: ```<img alt="Visual Studio App Center Releases" src="https://img.shields.io/visual-studio-app-center/releases/version/MyOwner/MyApplication/MyToken">```


where
"MyOwner" is my organisation name
"MyApplication" is my app name
"MyToken" is my token

To reiterate, this was generated from [Here](https://shields.io/badges/visual-studio-app-center-releases)

thanks

**Comment by chris48s** [comment]
OK, so here's an example of a known working badge

https://img.shields.io/visual-studio-app-center/releases/version/jct/test-fixed-android-react/8c9b519a0750095b9fea3d40b2645d8a0c24a2f3

The code you've flagged is fine as it stands. `short_version: shortVersion` assigns the value corresponding to the JSON key `short_version` to the JS variable `shortVersion`, so that isn't the issue.

Looking over the JSON response you posted, I think the reason the response for your project is failing validation is because of

```
...
"short_version": ""
...
```

We're rejecting it because of the empty `short_version`.

I guess firstly, do you have any idea why that is the case for your project?

I don't really have any handle on Visual Studio App Center as a service, but I guess what we should do here is accept an empty `short_version` in the schema, and if it is missing, then just display the value of `version` instead. Does that seem right?

**Comment by steve09014098** [comment]
Hi

Agreed that seems reasonable.

I have searched the docs for App Center and I cannot find any reference to a Short version.

I assume that it is something that can be auto filled if you upload the right type of project.

I push a release to app center using a yaml build pipeline who only gives me the option to set the version which I do.

I think your proposal is good, show the short version as it currently is, but if the short version is null or empty then revert to the long version.

Thanks for your support thus far!
